src/ts/miner.ts

```diff
@@ -0,0 +1,153 @@
+          ),
+        );
+        assert(value.length > 0 && value.length <= 32);
+
+        const opcode = PUSH1_OPCODE + value.length - 1;
+        return [opcode, ...value];
+      }
+    })
+    .reduce((bytecode, opcodes) => [...bytecode, ...opcodes], []);
+
+  return ethers.utils.hexlify(bytecode);
+}
+
+export const MINER_PAYOUT_CODE = assemble(`
+  ;; This contract just forwards the received value to the coinbase address.
+
+  ;; Start by pushing the required 'CALL' parameters to the stack in the correct
+  ;; order:
+  ;; First, push the return data length and and offset. Since we don't care
+  ;; about the return data at all, we can just specify 0's for these two
+  ;; parameters. We use 'RETURNDATASIZE' to push 0's onto the stack since its
+  ;; guaranteed to be 0 before doing any calls. This saves us 1 gas and 1
+  ;; byte of code over 'PUSH1 0x00'
+  RETURNDATASIZE ;; PUSH 0
+  RETURNDATASIZE ;; PUSH 0
+
+  ;; Second, push the calldata length and offset, again 0's because we don't
+  ;; want to provide any calldata. We use 'RETURNDATASIZE' for 0's for the same
+  ;; reason as above.
+  RETURNDATASIZE ;; PUSH 0
+  RETURNDATASIZE ;; PUSH 0
+
+  ;; Third, push the 'CALL' value for the transaction to the coinbase address.
+  ;; This should be whatever value was used to call this contract.
+  CALLVALUE
+
+  ;; Forth, push the address to 'CALL', that is the recipient of this transfer
+  ;; transaction. This is the coinbase address.
+  COINBASE
+
+  ;; Fifth, push the gas to use for the transaction. For simplicity and gas
+  ;; efficiency, just forward all remaining gas.
+  GAS
+
+  ;; Finally, execute the transaction for transferring the Ether.
+  CALL
+
+  ;; Now, our stack should only have a single value on it, which is the result
+  ;; of the previous call - 1 for success and 0 for failure. We can check for
+  ;; failure by jumping to an invalid code location if the success is zero.
+  ;; Jumping to an invalid code location causes an 'INVALID' trap to be raised,
+  ;; causing the call to this contract to fail.
+  ISZERO
+  ADDRESS ;; Random invalid location - costs 2 gas and 1 byte of code
+  JUMPI
+
+  ;; If we got here, that means that the call was successful because we did not
+  ;; jump to an invalid address. So just halt execution, returning no data.
+  STOP
+`);
+
+export const MINER_PAYOUT_CREATION_CODE = assemble(`
+  ;; This is the creation code for the code to forward transferred value to the
+  ;; block coinbase address.
+
+  ;; Creation code works by writing and returning code from memory, so lets
+  ;; start by writing the bytecode assembled above into memory at address 0 and
+  ;; then returning. Once again we use 'RETURNDATASIZE' to push 0's onto the
+  ;; stack.
+
+  ;; First push the data that we want to write to memory, i.e. the code. Note
+  ;; that we right-pad the bytes so that the first byte of code that is written
+  ;; to memory is at offset 0.
+  PUSH ${ethers.utils.hexlify(MINER_PAYOUT_CODE).padEnd(66, "0")}
```

Comment:
Nice way to avoid using `CODECOPY` and avoiding to compute bytecode locations. :smile: